### PR TITLE
Add TypeScript support

### PR DESCRIPTION
**Pull request by hbiede**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+declare module 'click-should-be-intercepted-for-navigation' {
+    declare function eventIsModifiedByKeyPress(event: Event): boolean;
+    declare function isLeftClick(event: Event): boolean;
+
+    function shouldInterceptEvent(event: Event): boolean;
```
Comment (TehShrike) [inline index.d.ts:5]: hmm, I wonder if these should be `MouseEvent` instead of `Event`?

File: package.json [hunk 2]
```diff
@@ -1,8 +1,9 @@
 {
   "name": "click-should-be-intercepted-for-navigation",
-  "version": "1.0.0",
+  "version": "1.1.0",
```
Comment (TehShrike) [inline package.json]:
```suggestion
  "version": "1.0.0",
```

the version bump happens automatically as part of my publish script

File: index.d.ts (hunk 1 above)
Comment (hbiede) [reply index.d.ts:5]: Up to you

Comment (TehShrike) [reply index.d.ts:5]: If it would be more correct to expect `MouseEvent`, then that's what I'd like 😅

Comment (hbiede) [reply index.d.ts:5]: `Event`s are tough without knowing every way it's utilized. I would be a little concerned being too strict if it's going to be attached to a component that types its output as `Event`

Comment (TehShrike) [reply index.d.ts:5]: Fine by me 👍

**Comment by TehShrike** [comment]
Published as 1.1.0

**Pull request merged**